Simplify linter setup

walles opened the pull request:
* Bump linter to support generics
* Enable all default linters
* Enable some named linters that were already used
* Remove lots of unused suppressions
* Handle some reported issues

walles commented on .golangci.yml:
```diff
@@ -3,60 +3,16 @@ run:
 linters-settings:
   govet:
     check-shadowing: true
-  gocyclo:
-    min-complexity: 10
-  dupl:
-    threshold: 100
```
These were explicitly disabled, no need to configure them.

alecthomas commented:
The existing configuration is (largely) deliberate, in particular enable-all.

If there's some specific change you'd like to address, please send through a smaller PR.

walles commented:
The inspiration for this PR was that the linter prints this in each CI run (example: https://github.com/alecthomas/chroma/actions/runs/6494594158/job/17638176581):
```
level=warning msg="[runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
level=warning msg="[runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner. Replaced by exportloopref."
level=warning msg="[runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner. Replaced by revive."
level=warning msg="[runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner. "
level=warning msg="[runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
level=warning msg="[runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused."
level=warning msg="[linters_context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
level=warning msg="[linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
level=warning msg="[linters_context] wastedassign is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
```
